substrate/frame/indices/src/lib.rs

```diff
@@ -62,8 +76,14 @@ pub mod pallet {
 			+ Copy
 			+ MaxEncodedLen;
 
+		/// The hold reason for this pallet.
+		type RuntimeHoldReason: From<HoldReason>;
+
 		/// The currency trait.
-		type Currency: ReservableCurrency<Self::AccountId>;
+		type NativeBalance: Inspect<Self::AccountId>
```

Comment:
I think we should keep the old name, (or otherwise rename to just `Balance`, nothing is forcing a concept of `native` here.)